pkg/virt-controller/watch/vm.go

```diff
@@ -1111,6 +1120,15 @@ func isSetToStart(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualMachineInstance)
 	}
 }
 
+func (c *VMController) cleanupRestartRequired(vm *virtv1.VirtualMachine) error {
```

Comment:
This can be dropped.

Reply:
Hey, I believe we can simplify this and just drop the whole `cleanupRestartRequired`. 

First of all, I see that the VM is not a copy in most if not all cases, leading to corruption. 
Second of all, there are these cases:
1. VM is starting or is recovering, the start creates a new revision, and in the `sync` the `addRestartRequiredIfNeeded` will just handle this.
2.  In case of a stop, do we care about the condition? If so we can enhance `addRestartRequiredIfNeeded` to just drop the condition in case of no revision.

Reply:
If the VM is stopped, I do not think we should care about the `RestartRequired`. On the next restart, it would disappear.

If we insist on keeping this, we need to make `DeepCopy` before we do `vmConditionManager.RemoveCondition(vm, virtv1.VirtualMachineRestartRequired)` and then return the `vm` for the condition to be removed by status update.

Reply:
I will not disappear on restart if we don't remove it. I added the missing DeepCopy, good catch! Thanks